File: week9/FirstLineRead/FileUtils.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class FileUtils
{
    /**
     * Ez a metódus egy fájlból soronként olvassa be az adatokat, és azokat egy listában tárolja.
     *
     * @param fileName A fájl neve, amelyből az adatokat be kell olvasni.
     * @return A fájl sorainak listája.
     */
    public static List<String> readLines(String fileName)
    {
        // A sorokat tároló lista inicializálása.
        List<String> lines = new ArrayList<String>();

        try
        {
            String line; // Egy változó az egyes sorok tárolására.

            /**
             * A fájl olvasásához a `BufferedReader`-t használjuk:
             * - A `FileReader` a fájl biteket olvas be.
             * - A `BufferedReader` a biteket sorokká alakítja.
             */
            BufferedReader br = new BufferedReader(new FileReader(fileName));

            // Az összes sor beolvasása a fájlból.
            while ((line = br.readLine()) != null)
            {
                // A beolvasott sor hozzáadása a listához.
                lines.add(line);
            }

            // Az olvasás után a fájlt lezárjuk, hogy erőforrásokat szabadítsunk fel.
            br.close();
        }
        catch (FileNotFoundException e)
        {
            /**
             * Ha a fájl nem található, akkor egy kivételt dobunk:
             * - Ezzel biztosítjuk, hogy a program ne folytassa a működését egy hiányzó fájllal.
             */
            e.printStackTrace();
            System.exit(1); // Kilépés hibaállapottal.
        }
        catch (IOException e)
        {
            /**
             * Ha egyéb bemeneti/kimeneti hiba történik:
             * - Például fájl olvasási problémák.
             */
            e.printStackTrace();
            System.exit(2); // Kilépés másik hibaállapottal.
        }

        return lines; // Az összegyűjtött sorok visszaadása.
    }

    /**
     * Ez a metódus a fájl első sorát adja vissza.
     *
     * @param filename A fájl neve, amelyből az adatokat be kell olvasni.
     * @return A fájl első sora.
     */
    public static String readFirstLine(String filename) {
        // A `readLines` metódus által visszaadott lista első elemét kérjük le.
        return readLines(filename).get(0);
    }

    /**
     * Ez a metódus a fájl első sorát számjegyek listájává alakítja.
     *
     * @param filename A fájl neve, amelyből az adatokat be kell olvasni.
     * @return A fájl első sorának számjegyei listaként.
     */
    public static List<Integer> readFirstLineAsListOfDigits(String filename) {
        // Az első sor beolvasása szövegként.
        String first = readFirstLine(filename);

        // Egy lista inicializálása az egyes számjegyek tárolására.
        List<Integer> result = new ArrayList<>();

        // A szöveg karakterenkénti feldolgozása.
        for (char c : first.toCharArray()) {
            // A karakterek számértékké alakítása és a lista feltöltése.
            result.add(Character.getNumericValue(c));
        }

        return result; // A számjegyek listájának visszaadása.
    }
}

File: week13/Practice/SolveTask4/FileUtils.cs
using System;
using System.IO;                    // A fájlkezeléshez szükséges névtér
using System.Collections.Generic;   // A List típus használatához szükséges névtér

namespace SolveTask4
{
    static class FileUtils
    {
        /*
         * A ReadLines metódus egy fájlt olvas be soronként és egy List<string>-et ad vissza, 
         * amely tartalmazza a fájl összes sorát.
         * 
         * Bemenet: 
         *      - fileName: A megnyitandó fájl neve, amelyet olvasni szeretnénk.
         * Kimenet:
         *      - Egy List<string> típusú lista, amely a fájl sorait tartalmazza.
         */
        public static List<string> ReadLines(string fileName)
        {
            var lines = new List<string>();     // Egy üres lista létrehozása, amelyben tárolni fogjuk a fájl sorait.

            // A StreamReader segítségével megnyitjuk a fájlt és olvassuk soronként.
            using (var f = new StreamReader(fileName))
            {
                string? line;                   // A fájl aktuális sorának tárolására szolgáló változó.
                
                // A ReadLine metódus segítségével olvassuk a fájl sorait addig, amíg el nem érjük a fájl végét (null).
                while ((line = f.ReadLine()) != null)
                {
                    lines.Add(line);            // A beolvasott sor hozzáadása a lista végéhez.
                }
            }

            // A fájl összes sorát tartalmazó lista visszaadása.
            return lines;
        }
    }
}
